src/main/java/ru/hogwarts/school/service/FacultyService.java

```diff
@@ -20,26 +20,38 @@ public FacultyService (FacultyRepository facultyRepository)
 
     public Faculty addFaculty(Faculty faculty)
     {
+        if(faculty == null)
+            return null;
+
         facultyRepository.save(faculty);
         return faculty;
     }
 
-    public Faculty getFaculty(long id)
+    public Faculty getFaculty(Long id)
     {
+        if(facultyRepository.findById(id).isEmpty())
+            return null;
         return facultyRepository.findById(id).get();
     }
 
     public Faculty removeFaculty(long id)
     {
-        Faculty tmpFacl = facultyRepository.findById(id).get();
+        if(facultyRepository.findById(id).isEmpty())
+            return null;
+        Faculty tmp = facultyRepository.findById(id).get();
         facultyRepository.deleteById(id);
```

Comment:
```suggestion
        facultyRepository.findById(id)
            .map(faculty -> acultyRepository.delete(id))
            .orElse(null);
        ;
```
вот так надо обработать